lib/ofp-util.c

```diff
@@ -1928,11 +1868,21 @@ ofputil_encode_update_map(enum ofp_version ofp_version,
     msg->map_id = htons(map);
     msg->key_size = htonl(key_size);
     msg->value_size = htonl(value_size);
+    msg->nb_elems = htonl(nb_elems);
 
-    ofpbuf_put(request, key, key_size);
-    ofpbuf_put(request, value, value_size);
-    ofpmsg_update_length(request);
+    void *data = malloc(length);
+    int offset = 0;
+    for(int i = 0; i < nb_elems; i++) {
+        memcpy(data + offset, key, key_size);
+        offset += key_size;
+        memcpy(data + offset, value, value_size);
+        offset += value_size;
+    }
```

Comment:
I think we can avoid the above `malloc`, `memcpy`s, and `free` by using `ofpbuf_put` directly. If you want to avoid `ofpbuf_put` reallocating its buffer several times, you can instead use [`ptr = ofpbuf_put_uninit()`](https://github.com/Orange-OpenSource/oko/blob/11208312300f78a015a1f3d13762d01b38ca5507/lib/ofpbuf.c#L358-L369) then copy data directly to `ptr`.